TCD-2905: Coroner component and Morgue page (#2262)
saucelabs/sauce-docs: docs/error-reporting/advanced/coroner/client-usage.md

---
id: client-usage
title: Coroner Client - Usage
sidebar_label: Client Usage
description: Interaction with the coronerd object store occurs either through the web console or the command-line tools.
---

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';
import useBaseUrl from '@docusaurus/useBaseUrl';

> Interaction with the coronerd object store can be done through the web console or the command-line tools. This section focuses on the coroner tool, which is a command-line client for interacting with the object store. It is easy to use but allows for complex queries and integrates easily into your terminal development environment.

Author: ```suggestion
Interaction with the coronerd object store can be done through the web console or the command-line tools. This section focuses on the coroner tool, a command-line client for interacting with the object store. It is straightforward but allows for complex queries and integrates efficiently into your terminal development environment.
```

## Initial Setup

### Coroner 0.12.0 or Newer

> After a user has been added to the object store, they must log into a server to interact with it.

Author: ```suggestion
After adding a user to the object store, the user must log into a server to interact with it.
```

To avoid passing commonly used options to `coroner list` and other `coroner` sub-commands for every invocation, you can use the default configuration block in your configuration file to specify certain options to always be set. Here's an example configuration:

```shell
[universe]
name=company
read=127.0.0.1:4097

[default]
list.frequency=environment,tag,dc,collection
list.expand=environment,tag,dc,collection,application,hostname
list.sort=hostname
list.instances=3
```

This specifies that whenever the `list` sub-command is invoked, it should have `--frequency=environment,tag,dc,collection`, `--expand=environment,tag,dc,collection,application,hostname`, `--sort=hostname`, and `--instances=3` prepended to the list of options supplied to the command-line.

The general form of the default configuration block is `.=`. For example, `X.Y=Z` specifies that if `coroner X` is specified, it should be interpreted as if it were `coroner X --Y=Z`.

## Examples

Here are some common use-cases:

### Sort Groups by Recent Activity

The `-R` option allows you to sort groups by the date of recent activity. For example: `coroner list project -R`.

### List All Ativity in The Last Week

> The `--age` option can be used to filter groups within a recent time window. For example, `coroner list project --age=1w` lists all groups and faults with activity in the past week.

Author: ```suggestion
The `--age` option can be used to filter groups in a recent time window. For example, `coroner list project --age=1w` lists all groups and faults with activity in the past week.
```
